benchmarks/model/mlx/cnn.py

```diff
@@ -0,0 +1,93 @@
+# Copyright © 2023-2024 Apple Inc.
+
+import types
+from typing import List, Tuple
+
+import mlx.core as mx
+import mlx.nn as nn
+import numpy as np
+from mlx.utils import tree_flatten
+
+from pfl.metrics import Weighted
+
+
+def maxpool2d(x, pool_size=2, stride=2):
+    batch, height, width, channels = x.shape
+    out_height = (height - pool_size) // stride + 1
+    out_width = (width - pool_size) // stride + 1
+    x = x.reshape(batch, out_height, stride, out_width, stride, channels)
+    # TODO: this is avg pooling, max doesn't work right now:
+    # https://github.com/ml-explore/mlx/issues/1234
```

Comment:
This is fixed in https://github.com/ml-explore/mlx/pull/1237, so we will need to monitor which mlx version has this change, and update here.

Reply:
looks like it was released in https://github.com/ml-explore/mlx/pull/1239 0.15.2, I will update